pennylane_lightning/core/src/measurements/MeasurementsBase.hpp

```diff
@@ -228,6 +228,166 @@ template <class StateVectorT, class Derived> class MeasurementsBase {
+        size_t obs_term_idx = 0;
+        for (const auto &coeff : coeffs) {
+            std::vector<size_t> shot_range = {};
+            auto obs_samples =
+                measure_with_samples(obs, num_shots, shot_range, obs_term_idx);
+            PrecisionT expval_per_term =
+                std::accumulate(obs_samples.begin(), obs_samples.end(), 0.0);
+            auto term_mean = expval_per_term / obs_samples.size();
+
+            result +=
+                coeff * coeff *
+                (1 - term_mean *
+                         term_mean); //`1` used here is because Eigenvalues for
+                                     // Paulis, Hadamard and Identity are {-1,
+                                     // 1}. Need to change based on eigen values
+                                     // when add Hermitian support.
+            obs_term_idx++;
+        }
+        return result;
+    }
+
+    /**
+     * @brief Probabilities of each computational basis state for an observable.
+     *
+     * @param obs An observable object.
+     *
+     * @return Floating point std::vector with probabilities.
+     * The basis columns are rearranged according to wires.
+     */
+    auto probs(const Observable<StateVectorT> &obs) {
+        PL_ABORT_IF(
+            obs.getObsName().find("Hamiltonian") != std::string::npos,
+            "Hamiltonian and Sparse Hamiltonian do not support samples().");
+        std::vector<size_t> obs_wires;
+        std::vector<size_t> identity_wires;
+        auto sv = _preprocess_state(obs, obs_wires, identity_wires);
+        Derived measure(sv);
+        return measure.probs(obs_wires);
+    }
+
+    /**
+     * @brief Return samples drawn from eigenvalues of the observable
+     *
+     * @param obs The observable object to sample
+     * @param num_shots Number of shots used to generate samples
+     *
+     * @return samples of eigenvalues of the observable
+     */
+    auto sample(const Observable<StateVectorT> &obs, const size_t &num_shots)
+        -> std::vector<PrecisionT> {
+        PL_ABORT_IF(
+            obs.getObsName().find("Hamiltonian") != std::string::npos,
+            "Hamiltonian and Sparse Hamiltonian do not support samples().");
+        std::vector<size_t> obs_wires;
+        std::vector<size_t> identity_wires;
+        std::vector<size_t> shot_range = {};
+        size_t term_idx = 0;
+
+        return measure_with_samples(obs, num_shots, shot_range, term_idx);
+    }
+
+    /**
+     * @brief Return the raw basis state samples
+     *
+     * @param num_shots Number of shots used to generate samples
+     *
+     * @return the raw basis state samples
+     */
+    auto sample(const size_t &num_shots) -> std::vector<size_t> {
+        Derived measure(_statevector);
+        return measure.generate_samples(num_shots);
+    }
+
+    /**
+     * @brief Groups the eigen values of samples into a dictionary showing
+     * number of occurences for each possible outcome with the number of shots.
```

Comment:
Just to confirm, but the returned dictionary would lack entries for eigenvalues with 0 occurrence right?